### ISSUE TITLE
write training / verification functionality

### ISSUE THREAD

**Issue by AreopagX** [reinhud/real-time-hand-gesture-detection#3]
- [x] forward passes can read values from our dataset class
- [x] forward passes generate outputs
- [x] loss is computed
- [x] optimizer step is called

**Comment by AreopagX** [comment]
suggestion: use pytorch lightning https://lightning.ai/docs/pytorch/stable/starter/introduction.html
